### ISSUE TITLE
php artisan migrate - errors creating trigger

### ISSUE THREAD

**Issue by eleftrik** [yajra/laravel-oci8#201]
When I run 'php artisan migrate', I get this output:

```
Migration table not found.
Migration table created successfully.


  [Illuminate\Database\QueryException]
  Error Code    : 24344
  Error Message : ORA-24344: success with compilation error
  Position      : 29
  Statement     :
              create trigger users_id_trg
              before insert on users
              for each row
                  begin
              if :new.id is null then
                  select users_id_seq.nextval into :new.id from dual;
              end if;
              end;
  Bindings      : []
   (SQL:
              create trigger users_id_trg
              before insert on users
              for each row
                  begin
              if :new.id is null then
                  select users_id_seq.nextval into :new.id from dual;
              end if;
              end;)



  [Yajra\Pdo\Oci8\Exceptions\Oci8Exception]
  Error Code    : 24344
  Error Message : ORA-24344: success with compilation error
  Position      : 29
  Statement     :
              create trigger users_id_trg
              before insert on users
              for each row
                  begin
              if :new.id is null then
                  select users_id_seq.nextval into :new.id from dual;
              end if;
              end;
  Bindings      : []
```

After seeing this error, 'Migration' and 'Users' tables are created, 'users_id_seq' also; 'users_id_trg' is created, but marked with "invalid" status (I'm using TOAD, this trigger is listed but with a red cross), 
TOAD shows me these errors:

PLS-00103: Encountered the symbol "" - Line 1 - Column 6
PLS-00103: Encountered the symbol "" - Line 2 - Column 36
PLS-00103: Encountered the symbol "" - Line 3 - Column 68

If I run 'create trigger...' manually, I don't get any error.

Thank you
### System details
- Operating System: Linux Debian 8
- PHP Version:  7.0.9-1~dotdeb+8.1 (cli) ( NTS )
- Laravel Version: 5.3.4
- Yajra/Laravel-oci8 Version: 5.3.1

**Comment by yajra** [comment]
Duplicate - related #202.

**Comment by eleftrik** [comment]
Hi @yajra, I re-created the user and gave it all the grants, as you suggested in [#202](https://github.com/yajra/laravel-oci8/issues/202), but I get the same error about the trigger.

Here two screenshots:

php artisan error
https://postimg.org/image/7b6ja0yyz/

trigger status as seen on TOAD
https://postimg.org/image/n51meorgj/

Thank you.

**Comment by eleftrik** [comment]
@yajra, I think I found a workaround.

If I edit these files:

vendor/yajra/laravel-oci8/src/Oci8/Schema/Sequence.php
vendor/yajra/laravel-oci8/src/Oci8/Schema/Trigger.php

and I remove line feeds from every _create trigger, create sequence, drop trigger, drop sequence_ statements (so every DDL statement is on a single line), everything works as expected (I can run `php artisan migrate` and `php artisan migrate:refresh`)

Sorry for my trivial exposition, I hope it's clear enough.

Thank you

**Comment by mstaack** [comment]
which oracle version are you using? cant reproduce with fresh installation (oracle-homestead)
maybe this gives more infos about the error: http://www.dba-oracle.com/t_ora_24344_success_with_compilation_error.htm

**Comment by eleftrik** [comment]
@mstaack Oracle 10.2.0.4.0

**Comment by eleftrik** [comment]
Hi @yajra, do you have any update?
I'm experiencing the same error while calling, for example, ```Schema::dropIfExists('table')```.

Did you read about my "[workaround](https://github.com/yajra/laravel-oci8/issues/201#issuecomment-243372582)" ?

Thank you

**Comment by yajra** [comment]
@eleftrik sorry for late response. Just saw your workaround and this might be an OS specific issue? What OS are you using? 

Anyways, if it works on your side, can you please send a PR for the changes you've made? Thanks!

**Comment by yajra** [comment]
I just made a quick test and seems to work fine with me. Below is the test migration I used:

```php
use Illuminate\Support\Facades\Schema;
use Illuminate\Database\Schema\Blueprint;
use Illuminate\Database\Migrations\Migration;

class DropIfExists extends Migration
{
    /**
     * Run the migrations.
     *
     * @return void
     */
    public function up()
    {
        Schema::create('demos', function (Blueprint $table) {
           $table->increments('id');
        });
    }

    /**
     * Reverse the migrations.
     *
     * @return void
     */
    public function down()
    {
        Schema::dropIfExists('demos');
    }
}
```

**Comment by yajra** [comment]
`php artisan migrate:refresh` also works well for me. Can you please paste your migration file?

**Comment by eleftrik** [comment]
Hi @yajra, thanks for your kind reply.

I'm developing on a Windows workstation running a Vagrant box (Debian 8); the Oracle database is on a production server (OS: AIX).

I think the problem is not in the migration file, because
1) it runs with no problem on a MySql database
2) if I apply the same workaround to `compileDropIfExists` method (vendor/yajra/laravel-oci8/scr/Oci8/Schema/Grammars/OracleGrammar.php), `php artisan migrate:refresh` works perfectly.

Anyway, my migration is:

```PHP
<?php

use Illuminate\Support\Facades\Schema;
use Illuminate\Database\Schema\Blueprint;
use Illuminate\Database\Migrations\Migration;

class CreateIpscanNetworksTable extends Migration
{
    /**
     * Run the migrations.
     *
     * @return void
     */
    public function up()
    {
        Schema::create('ipscan_subnet', function (Blueprint $table) {
            $table->unsignedInteger('id')->primary();
            $table->string('subnet');
            $table->string('description');
            $table->unsignedInteger('first_ip');
            $table->unsignedInteger('last_ip');
            $table->unsignedInteger('order_by')->nullable();
            $table->boolean('active')->default(true);
            $table->nullableTimestamps();
            $table->softDeletes();
        });
    }

    /**
     * Reverse the migrations.
     *
     * @return void
     */
    public function down()
    {
        Schema::dropIfExists('ipscan_subnet');
    }
}
```

`Schema::dropIfExists()` raises an exception like this:

```
 [Yajra\Pdo\Oci8\Exceptions\Oci8Exception]                                                                                                                
 Error Code    : 6550                                                                                                                                     
 Error Message : ORA-06550: line 1, column 15:                                                                                                            
 PLS-00103: Encountered the symbol "" when expecting one of the following:                                                                                
    begin function package pragma procedure subtype type use                                                                                              
    <an identifier> <a double-quoted delimited-identifier> form                                                                                           
    current cursor                                                                                                                                        
 The symbol "" was ignored.                                                                                                                               
 Position      : 14                                                                                                                                       
 Statement     : declare c int;                                                                                                                           
         begin select count(*) into c from user_tables where table_name = upper('ipscan_subnet'); if c = 1 then execute immediate 'drop table ipscan_subne
 t'; end if; end;                                                                                                                                         
 Bindings      : []                                                                                                                                       
```

As soon as there is a line separator in the SQL statement, I got errors like this.

I didn't make any PR because mine is a dirty workaround :)

**Comment by eleftrik** [comment]
@yajra, maybe I found the cause.
As I wrote, I'm developing on a Windows workstation and I use PHPStorm IDE. This one shows that the line separator in CRLF (I think GIT handles the conversion to LF when pushing to the remote repository).

If I set "your" files to use **LF** as line separator, everything works perfectly.

**Comment by elib0** [comment]
In the trigger, after the `before` instance, need to add this
```php
referencing new as new old as old
```
I fixed this, but need something more. When you use a schema prefix, the trigger don't add it, the same with the sequences.

**Comment by yajra** [comment]
@elib0 schema prefix was fixed via #301 . Thanks!
